pkg/statsd/lexer.go

```diff
@@ -110,6 +110,8 @@ func lexSpecial(l *lexer) stateFn {
 	default:
 		l.pos--
 		l.m = l.metricPool.Get()
+		// Pull the tags from the metric, because it may have a buffer we can reuse.
+		l.tags = l.m.Tags
```

Comment:
Not too sure this is the best way.  Could have another pool, but (despite giving them their own package), I kind of want to avoid the proliferation of pools.

Reply:
"may have an empty buffer" (because if a buffer exists it was cleared in m.Reset() before the pool gave it to us, but let's not make the next person go looking for that)